packages/protobuf/src/private/reflect.ts

```diff
@@ -67,16 +70,45 @@ export function clearField(
         target[localName] = {};
         break;
       case "enum":
-        target[localName] = implicitPresence ? field.T.values[0].no : undefined;
+        if (implicitPresence) {
+          target[localName] = field.T.values[0].no;
```

Comment:
No way to have `scalarZeroValue` compute this for enums? It would be nice to consolidate this case with the one below given how much repetition there is. (Just a thought; not a blocking a comment.)

Reply:
I agree that it would be nice to avoid the repetition here, but it would add enough complication for `scalarZeroValue` and the [ScalarValue](https://github.com/bufbuild/protobuf-es/blob/89777bb0a6e84c8440be0f23bc62a27071683fc9/packages/protobuf/src/scalar.ts#L86) type (both parts of the public API), that it wouldn't be a net improvement overall.